src/blog/2024/04/understanding-node-flow-global-environment-variables-in-node-red.md

```diff
@@ -0,0 +1,233 @@
+
+### Exploring Global variables
+
+Global variables in Node-RED are accessible to function, change, inject, switch nodes within a given Node-RED instance. They serve as a centralized storage point for data that needs to be accessed across different parts of an application. This is especially useful when you want to share data across multiple flows or tabs within the same Node-RED instance.
+
+For example, in a home automation system with flows for lighting, security, and climate control, global variables can store user preferences or system settings that all flows can access and update. This allows for consistent behaviour across the entire system.
+
+#### Initiating/setting Flow variable
+
+We can set Global variables using the function and change node.
+
+In the change node, you can set it as shown in the below image
+
+!["Screenshot showing how to set global variable using the change node"](./images/variables-in-node-red-setting-global-variable-using-change-node.png "Screenshot showing how to set global variable using the change node"){data-zoomable}
+
+To initiate a Flow variable in the function node, you will have to use the global`s set method like the below in the function node:
+
+```javascript
+global.set('variableName', value);
+```
+
+#### Retrieving Global variable
+
+Retrieving global variables using change, inject, and switch nodes is quite similar. You simply need to select the "global" option and enter the variable name in the input field. Below is an image showing how you can retrieve global variables using the change node:
+
+!["Screenshot showing how to retrieve global variables using the change node"](./images/variables-in-node-red-retrieving-global-variable-using-change-node.png "Screenshot showing how to retrieve global variables using the change node"){data-zoomable}
+
+To retrieve a global variable using the function, use the global's `get` method like the below in the function node:
+
+```javascript
+global.get('variableName');
+```
+
+#### Deleting Global variables
+
+To delete the global variables again you can use both the "context data" tab and the change node. I have shown in the below image how you can delete global variables using the change node:
+
+!["Screenshot showing how to delete global variable using the change node"](./images/variables-in-node-red-deleting-global-variable-using-change-node.png "Screenshot showing how to delete global variable using the change node"){data-zoomable}
+
+### Exploring Environmental variables
+
+Environment variables are specifically used for storing sensitive configuration data, such as API keys or database credentials, ensuring this information isn't directly exposed within your flows. In Node-RED you can set environment variables at flow and global level.
```

Comment:
Maybe link to: https://flowfuse.com/blog/2023/01/environment-variables-in-node-red/

Reply:
Yes, I've already linked that in the global-level environment variable section.